dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/OmegaRef/GeneratorPV.mo

```diff
@@ -65,8 +63,10 @@ equation
   end when;
 
   if running.value then
+    URefPu.value = UPu + LambdaPu * (QGenRefPu + T * der(QGenRefPu));
     QGenPu = if qStatus == QStatus.AbsorptionMax then QMaxPu else if qStatus == QStatus.GenerationMax then QMinPu else QGenRefPu;
   else
+    QGenRefPu = 0;
```

Comment:
Does the disconnection still fail when you keep UPu outside the running loop and you have this modification?

Reply:
The problem is not when the generator is disconnected but when the line that connects the generator to the network is disconnected. In that case, the bus is switched off and calculating UPu=abs(terminal.V) fails. 
This failure was encountered with DynaFlow (with the SignalN models) and putting UPu inside the if loops solves the problem. 
The change here for DynaWaltz GeneratorPV is to adapt QGenRef as UPu is now equal to 0.

Reply:
One solution could be to put UPu = abs(terminalV) in an if switchOffSignal1 loop instead of an if running loop. This way UPu = 0 only when the bus is switched off and not when only the generator is switched off

Reply:
One last thing, in GeneratorSynchronous.mo, the UPu calculation is put inside the if loop as well

Reply:
Yes it is inside the running loop for several models because we noticed this issue several times... I just wanted to make sure that we have in mind that it is not what we want to write if we really do separation between the modelling and the solving part. One day we can maybe have a closer look to it and check if it is normal to have a nan in the jacobian when that happens... But as I said, I will approve this PR because for now on we don't know how to fix this with a better way.

Reply:
Ok, thanks, we'll keep it in mind